### PR TITLE
Hardening

### PR DESCRIPTION
**Pull request by elrido**
* ADDED: Set CSP also as meta tag, to deal with misconfigured webservers mangling the HTTP header
* ADDED: Sanitize SVG preview, preventing script execution in instance context
* CHANGED: Upgrading libraries to: Showdown 2.0.3
* CHANGED: Drop support for attachment download in IE
* FIXED: Error when attachments are disabled, but paste with attachment gets displayed

### CODE REVIEW

File: js/privatebin.js [hunk 1]
```diff
@@ -2752,6 +2779,34 @@ jQuery.PrivateBin = (function($, RawDeflate) {
...
+                buf[i] = data.charCodeAt(i);
+            }
+            const blob = new window.Blob(
+                [buf],
+                {
+                    type: mimeType
+                }
+            );
+
+            // Get Blob URL
```
Comment (rugk) [inline js/privatebin.js:2805]:
```suggestion
            // Get blob URL
```

File: js/privatebin.js [hunk 2]
```diff
@@ -2761,44 +2816,42 @@ jQuery.PrivateBin = (function($, RawDeflate) {
...
             if (typeof fileName !== 'undefined') {
                 $attachmentLink.attr('download', fileName);
             }
 
-            me.handleBlobAttachmentPreview($attachmentPreview, blobUrl, mediaType);
+            // sanitize SVG preview
+            // prevents executing embedded scripts when CSP is not set and user
+            // right-clicks/long-taps and opens the SVG in a new tab - prevented
+            // in the preview by use of an img tag, which disables scripts, too
+            if (mimeType.match(/image\/svg/i)) {
```
Comment (rugk) [inline js/privatebin.js:2846]: Ah this also matches [mime types like])https://stackoverflow.com/questions/11918977/right-mime-type-for-svg-images-with-fonts-embedded#11930514) `image/svg+xml`, as it is not limited to start/end of string okay?

Though maybe limit the start as there is really nothing before?

```suggestion
            if (mimeType.match(/^image\/svg/i)) {
```

Also, a unit test should test this matching, for any elemnts beginning with `image/svg` (including the one with `+xml`)

**Review by rugk — CHANGES_REQUESTED** [review]

File: js/privatebin.js (hunk 2 above)
Comment (elrido) [reply js/privatebin.js:2846]: While it is highly unlikely that future types get added to include something that ends in "image", given that they only got 3 additions over the last 40 years, it doesn't hurt to be specific on that side.

For the purposes of this logic, we must regard the mimeType to be a input from an uncontrolled source (it is set by the paste creator), so we should match it as broadly as possible for sanitization purposes. Worst case scenario is that a non-SVG gets sanitized into something that doesn't render in the preview, but that is a fail-safe and IMO preferred over the opposite case.

Hence I think we should apply your suggestion to both this pattern as well as to the ones in handleBlobAttachmentPreview() further down. And we should relax the subtype side to match any svg-ish thing, even if non-standard. I'll share an improvement for your review, shortly.

Base types are documented in:
https://www.iana.org/assignments/media-types/media-types.xhtml

Comment (elrido) [reply js/privatebin.js]: Should be addressed in [75dc346](https://github.com/PrivateBin/PrivateBin/pull/906/commits/75dc346f0fbe81c843ab0aadbfc697ded33c14bd)

**Review by rugk — APPROVED** [review]

**Pull request merged**